materialize-boilerplate/transactions_stream.go

```diff
@@ -129,6 +190,11 @@ func (l *transactionsStream) storedHistoryHandler(delegate func(transactionsEven
 }
 
 func (l *transactionsStream) Send(m *pm.Response) error {
+	// Every time there is some activity, we send a signal to
+	// the dbt job trigger channel to postpone the trigger another 5 minutes
+	// we only want to trigger the job after minutes of inactivity by the connector
+	l.dbtJobTriggerCh <- true
```

Comment:
I think the dbt job triggering should only occur when the connector is sending an acknowledgement that it has committed some data, so down around line 205, prior to the `maybeDelayAcknowledgement`.

Regarding the edge case about the default 5 minute maximum transaction duration, I think it's pretty important that we aren't restricted by that and have to introduce any kind of explanations for how that interacts with dbt job triggering. Transactions can also be spread out longer than 5 minutes for various other reasons, like if load queries / processing loaded documents takes a long time, or if an actual commit to the destination takes a long time. I am also quite convinced that this dbt job scheduling should not be associated with the delays introduced by the sync frequency at all. Some materializations that support dbt job triggers don't have sync frequency configuration, sometimes users will set it to 0, and ultimately I think the entire ack delay mechanism [should probably live in the runtime](https://github.com/estuary/connectors/issues/1182), although the steps to get there aren't clear right now.

So here's what I'd propose, which I think builds on the async goroutine mechanism you've got here, and would not be constrained by any assumptions about how frequently commits are occuring:

Every time the materialization sends an acknowledgement but _before_ applying an acknowledgement delay, it should send a signal on `dbtJobTriggerCh`, for example between the current line 204 and 205.

Generally the `dbtJobTriggerHandler` would need to do this:
1. Trigger the dbt job right away if one hasn't been run in the current connector's running time. Note that this will cause a dbt job to be triggered the first time a connector starts up. But I actually think this is necessary to prevent cases where a delayed background dbt trigger never gets to run because the connector restarts before it does.
3. Trigger the dbt job right away if the last one was ran more than **dbtRunInterval** ago (I'd suggest hard-coding this at 30 minutes, and potentially make it configurable if users ask for it)
4. If the last dbt job ran within **dbtRunInterval** ago, schedule the next run **dbtRunInterval** after the last one ran, but...
5. ...if there is _already_ a job scheduled, don't do anything.

Specifically I think something like this would work, where the `dbtJobTriggerCh` has been modified to a `chan struct{}`:

```go
func (l *transactionsStream) dbtJobTriggerHandler(ctx context.Context, dbtRunInterval time.Duration) {
	var lastRun time.Time
	var nextRunScheduled atomic.Bool

	runJob := func() {
		lastRun = time.Now()
		if err := l.triggerDBTJob(); err != nil {
			log.WithField("err", err).Error("dbt job trigger failed")
		}
		nextRunScheduled.Store(false)
	}

	for {
		select {
		case <-l.dbtJobTriggerCh:
			if nextRunScheduled.Load() {
				// Another dbt job is set to run already.
			} else if time.Since(lastRun) >= dbtRunInterval {
				// It has been a long time since the last job ran; trigger it
				// now.
				runJob()
			} else {
				// The last job was ran recently. Don't run another job right
				// away, but make sure one does eventually get run.
				nextRunScheduled.Store(true)

				nextRunTime := lastRun.Add(dbtRunInterval)
				go func() {
					select {
					case <-time.After(time.Until(nextRunTime)):
						runJob()
					case <-ctx.Done():
						return
					}
				}()
			}
		case <-ctx.Done():
			return
		}
	}
}
```